android/views/src/main/kotlin/com/boswelja/ephemeris/views/CalendarPagerAdapter.kt

```diff
@@ -49,50 +48,106 @@ internal class CalendarPagerAdapter : InfinitePagerAdapter<CalendarPageViewHolde
 
     override fun onBindHolder(holder: CalendarPageViewHolder, page: Int) {
         val pageState = pageLoader!!.getPageData(page)
-        holder.bindDisplayRows(dayBinder!!, pageState)
+        holder.bindDisplayRows(pageLoader!!, dayBinder!!, pageState)
     }
 }
 
 internal class CalendarPageViewHolder(
     private val binding: CalendarPageBinding
-) : RecyclerView.ViewHolder(binding.root) {
+) : ViewHolder(binding.root) {
 
     private val inflater = LayoutInflater.from(itemView.context)
 
+    private val rowBindingCache = mutableListOf<CalendarRowBinding>()
+    private val dateCellViewHolderCache = mutableListOf<ViewHolder>()
+
+    private var boundDateCells = 0
```

Comment:
is this actually any different to
```kotlin
private val boundDateCells: Int
    get() = dateCellViewHolderCache.size
````
Seems a bit excessive to keep track of 2 variables for this? Or is it purely for calculating how many to drop when you're trimming?

Reply:
It's there to track which date holder to try look up from the cache, and it's reused when trimming yep